Blaze: Unit tests for BlazeCampaignCreationPreviewViewModel

hichamboushaba opened the pull request:
⚠️ Please don't merge until `trunk` is the base branch.

### Description
This PR adds unit tests for `BlazeCampaignCreationPreviewViewModel`.

### Testing instructions
Just green CI is enough.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Simplify model

hichamboushaba committed:
Remove duplicated space

hichamboushaba committed:
Add unit tests for the Blaze campaign preview

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/BlazeRepository.kt:
```diff
@@ -305,7 +302,6 @@ class BlazeRepository @Inject constructor(
         val productId: Long,
         val tagLine: String,
         val description: String,
-        val userTimeZone: String,
```
I removed this property from the model, we already handle passing the device's timezone ID at the `Store` level, and this also fixes an issue where we were accidentally using `displayName` instead of the timezone's `id`.

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewViewModel.kt:
```diff
@@ -259,7 +259,7 @@ class BlazeCampaignCreationPreviewViewModel @Inject constructor(
             startDate.formatToMMMdd()
         )
-        return "$totalBudgetWithCurrency,  $duration"
+        return "$totalBudgetWithCurrency, $duration"
```
I think this was just a typo, please correct me if I'm wrong.

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hichamboushaba committed:
Fix detekt issue

The formatting suggested by Detekt seems off, but I can't find a way to suppress the indentation errors.

hichamboushaba committed:
Update tests to align with the new models

0nko reviewed (approved):
LGTM 👍
